src/AppBundle/Controller/MembershipShiftExemptionController.php

```diff
@@ -18,30 +21,74 @@
  */
 class MembershipShiftExemptionController extends Controller
 {
+    /**
+     * Filter form.
+     */
+    private function filterFormFactory(Request $request): array
+    {
+        // default values
+        $res = [
+            "shiftExemption" => null,
+        ];
+
+        // filter creation ----------------------
+        $res["form"] = $this->createFormBuilder()
+            ->setAction($this->generateUrl('admin_membershipshiftexemption_index'))
+            ->add('shiftExemption', EntityType::class, array(
+                'label' => 'Motif',
+                'class' => 'AppBundle:ShiftExemption',
+                'choice_label' => 'name',
+                'multiple' => false,
+                'required' => false,
+            ))
+            ->add('filter', SubmitType::class, array(
+                'label' => 'Filtrer',
+                'attr' => array('class' => 'btn', 'value' => 'filtrer')
+            ))
+            ->getForm();
+
+        $res['form']->handleRequest($request);
+
+        if ($res['form']->isSubmitted() && $res['form']->isValid()) {
+            $res["shiftExemption"] = $res["form"]->get("shiftExemption")->getData();
+        }
+
+        return $res;
+    }
+
     /**
      * Lists all membershipShiftExemption entities.
      *
-     * @Route("/", name="admin_membershipshiftexemption_index", methods={"GET"})
+     * @Route("/", name="admin_membershipshiftexemption_index", methods={"GET","POST"})
      * @Security("has_role('ROLE_USER_MANAGER')")
      */
     public function indexAction(Request $request)
     {
+        $em = $this->getDoctrine()->getManager();
+        $filter = $this->filterFormFactory($request);
+        $findByFilter = array();
```

Comment:
```suggestion
        $findByFilter = $filter['filter'];
```

cf mon autre commentaire

Reply:
ca ne marche pas :thinking: 
je vais laisser tel quel, je me suis inspiré des filter form existants sur JobController & PeriodController, tu peux proposer une PR qui refactore ces filtres ?
et plus généralement, je ne suis pas fan du code nécessaire pour ces filterForm (on devrait utiliser la méthode GET + j'ai l'impression que c'est très verbeux avec ces if partout...)